netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -105,6 +128,59 @@ public void onGoAwayReceived(int lastStreamId, long errorCode, ByteBuf debugData
     });
   }
 
+  /**
+   * Sends a ping to the server.
+   */
+  void ping(final PingCallback callback, Executor executor) {
+    checkState(ctx != null);
+    Http2Ping newPing = null;
+    Stopwatch stopwatch = null;
+    ByteBuf buffer = null;
+    while (true) {
+      if (goAwayStatus != null) {
```

Comment:
I will note that goAwayStatus does not actually mark the (soon) end of the transport; the transport could remain alive for days after receiving a GOAWAY. The only thing that GOAWAY implies is that we will stop making new RPCs on the transport. The real source-of-truth is whether NettyClientTransport is terminated/the channel is inactive.

I don't know how much that matters for this feature, though.